recipes-config/systemd-resource-control/systemd-resource-control.bb

```diff
@@ -8,7 +8,12 @@ SRC_URI += " \
     file://LICENSE \
     file://connectivity.slice \
     file://ui.slice \
+    file://service.bluetooth.conf \
```

Comment:
Does it make sense to only add the file if DISTRO_FEATURES contains bluetooth, like:

`${@bb.utils.contains('DISTRO_FEATURES', 'bluetooth', 'file://service.bluetooth.conf', '', d)} 
`

Reply:
A bit of a similar question... is there a way to say "if package installed, include this override". 

As far as I can tell there is no variable one can query with bb.utils.contains that contains all packages that will be installed, or is there?

E.g. if `connectivity-manager` is removed, this recipe will have to be updated as well. Easy to miss. But if it is complicated to install overrides conditionally in this package, it is perhaps better to live with potentially missing removing them if certain packages are not installed...?

Reply:
> As far as I can tell there is no variable one can query with bb.utils.contains that contains all packages that will be installed, or is there?

Not that I'm aware of. I think you need to check this as part of a ROOTFS_POSTPROCESS_COMMAND.

```
my_postprocess_function() {
   echo "hello" > ${IMAGE_ROOTFS}/hello.txt
}
ROOTFS_POSTPROCESS_COMMAND += "my_postprocess_function; "
```

Reply:
> E.g. if `connectivity-manager` is removed, this recipe will have to be updated as well. Easy to miss. But if it is complicated to install overrides conditionally in this package, it is perhaps better to live with potentially missing removing them if certain packages are not installed...?

Yes, so perhaps it's better to have a rootfs postprocess command which can clean overrides

```
for all folders matching xxx.d in ${IMAGE_ROOTFS}/lib/systemd/system
  IF xxx.d EXISTS BUT NOT xxx THEN rm -f xxx.d
```

Reply:
Or perhaps just warn (or maybe even fail) if an override is detected for a service that is not installed ("X has been removed from dist => must update this recipe")?

Reply:
Keep in mind having an append/override for nonexistent service is not going to break anything.

Reply:
> Or perhaps just warn (or maybe even fail) if an override is detected for a service that is not installed ("X has been removed from dist => must update this recipe")?

Yes, a report would good.

Reply:
> Keep in mind having an append/override for nonexistent service is not going to break anything.

You are right, and a report, part of the rootfs postprocess, would highlight these inconsistency issues.